### PR TITLE
support sqlite plugin

### PR DESCRIPTION
**Pull request by broncha**
As per https://github.com/mWater/mwater-app/issues/639
I ended up adding the initialization logic to WebSQLDb itself.

sqlitePlugin.openDatabase could take a while to properly open the database, and the schema migrations can only be done when the database is open. 

autoselectLocalDb sets an option parameter `storage = sqlite` to activate the sqlite plugin.

### CODE REVIEW

File: src/WebSQLDb.coffee [hunk 1]
```diff
@@ -12,16 +12,54 @@ module.exports = class WebSQLDb
   constructor: (options, success, error) ->
```
Comment (grassick) [inline src/WebSQLDb.coffee]: Needs comments in the header on how it works and on the constructor on options.

File: src/WebSQLDb.coffee [hunk 2]
```diff
@@ -12,16 +12,54 @@ module.exports = class WebSQLDb
...
+              @db.transaction (tx) =>
+                tx.executeSql('''
+                  CREATE TABLE docs (
+                  col TEXT NOT NULL,
+                  id TEXT NOT NULL,
+                  state TEXT NOT NULL,
+                  doc TEXT,
+                  base TEXT,
+                  PRIMARY KEY (col, id));''', [], doNothing, ((tx, err) -> error(err)))
+                tx.executeSql("PRAGMA user_version = 2", [], doNothing, ((tx, err) -> error(err)))
```
Comment (grassick) [inline src/WebSQLDb.coffee]: Won't this set user_version to 2 even if the create table fails?

Comment (broncha) [reply src/WebSQLDb.coffee]: would it? both queries are within the same transaction

Comment (grassick) [reply src/WebSQLDb.coffee]: That's a good point... it would bomb out of the transaction.

**Pull request merged**